### PR TITLE
Added spoor id transform

### PR DESCRIPTION
**Pull request by Gkastrin**
This PR is for https://github.com/Financial-Times/spoor-enrichment/issues/289

@commuterjoy @adambraimbridge

### CODE REVIEW

**Comment by adambraimbridge** [comment]
This adds a `spoorId` property to `egest.annotations`, and doesn't change/remove any other instance of a Spoor ID value. 👍

**Comment by Gkastrin** [comment]
TODO we need to add the environment variable transform_spoor_id to heroku.

**Comment by commuterjoy** [comment]
Can you take a look at the transforms that extend the Transform class? It will remove dome of the boilerplate code if you switch to that format

**Comment by commuterjoy** [comment]
Eg, https://github.com/Financial-Times/spoor-enrichment/blob/master/server/transforms/time.js

File: tests/server/transforms/spoor-id.spec.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+'use strict';
+
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const spoorId = require('../../../server/transforms').spoorId;
+
+let e;
+
+describe.only('spoorId', () => {
+
```
Comment (adambraimbridge) [inline tests/server/transforms/spoor-id.spec.js]: We'll need to remove the ".only" here.

File: tests/server/transforms/spoor-id.spec.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			expect(spoorId).to.equal('12345-abcde');
+			done();
+		});
+	});
+
+	it('Should return a spoor id when supplied with a device spoor id', done => {
+
+		sinon.stub(e, 'pluck').onCall(0).returns('12345-abcde');
+
```
Comment (adambraimbridge) [inline tests/server/transforms/spoor-id.spec.js]: This might be better as `sinon.stub(e, 'pluck').withArgs('spoor-id').returns('12345-abcde');`

**Comment by Gkastrin** [comment]
@commuterjoy thanks for your comments, could you please have another look and let me know if you have any further suggestions

**Comment by commuterjoy** [comment]
@Gkastrin looks good - I changed a couple of things in 570bf30

In JavaScript lots of different things mean false, so you can use that to simplify the code a bit.r

https://developer.mozilla.org/en-US/docs/Glossary/Falsy

File: tests/server/transforms/spoor-id.spec.js [hunk 3]
```diff
@@ -10,48 +10,36 @@ let e;
...
 		sinon.stub(e, 'pluck').withArgs('device.spoor_id').returns('12345-abcde');
-
 		spoorId(e).then(spoorId => {
-
 			expect(spoorId).to.equal('12345-abcde');
 			done();
 		});
 	});
 
 	it('Should return an empty string if no spoor id was found', done => {
```
Comment (adambraimbridge) [inline tests/server/transforms/spoor-id.spec.js]: 'Should return `undefined` if no spoor id was found'

**Pull request merged**